lib/libprocstat/libprocstat.c

```diff
@@ -399,6 +405,53 @@ procstat_getc18n(struct procstat *procstat, struct kinfo_proc *kp,
 	return (-1);
 }
 
+int
```

Comment:
This should not require the caller to allocate an array and hope for the best. :)  It should return an allocated array of objects that the caller can free.  You can then call the sysctl twice, once to get the size estimate and a second time to populate it.  This requires fixing the sysctl to add the optimized path for querying the size I mentioned.